### PR TITLE
Add pyproject.toml

### PR DESCRIPTION
**Pull request by benjaminrwilson**
This PR adds a `pyproject.toml` file to consolidate styling tools such as `isort` and `black`.

- `pyproject.toml` was introduced in [PEP 518](https://www.python.org/dev/peps/pep-0518/) for build system requirements.
- Allows for a unified configuration file for a host of tools, e.g. `isort`, `black`, etc.

Each "block" within this file specifies a different type of options. For example, `isort` is written as:

```
[tool.isort]

known_first_party = "argoverse"
line_length = 120
multi_line_output = 3
profile = "black"
```

The options above should have a 1-to-1 correspondence with `isort.cfg`.

### CODE REVIEW

File: .isort.cfg [hunk 1]
```diff
@@ -1,5 +0,0 @@
-[settings]
```
Comment (tagarwal-argoai) [inline .isort.cfg:1]: Can you explain what the purpose of the file was before this change and why it has been removed?

Comment (tagarwal-argoai) [inline .isort.cfg:1]: Can you explain what the purpose of the file was before this change and why it has been removed?

File: argoverse/evaluation/detection/utils.py [hunk 2]
```diff
@@ -338,10 +338,7 @@ def rank(dts: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
     ranked_scores = scores[ranks]
 
     # Ensure the number of boxes considered per class is at most `MAX_NUM_BOXES`.
-    if ranked_dts.shape[0] > MAX_NUM_BOXES:
```
Comment (tagarwal-argoai) [inline argoverse/evaluation/detection/utils.py:341]: Can you change the base of this PR to point to the branch used in 227 as I am sure this change is not part of this PR?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[tool.isort]
```
Comment (tagarwal-argoai) [inline pyproject.toml:1]: Can you add a description to the PR on how to parse this file and what the different fields in this file indicate?

**Review by tagarwal-argoai — COMMENTED** [review]
Left a couple of clarifying questions.

File: pyproject.toml (hunk 3 above)
Comment (benjaminrwilson) [reply pyproject.toml:1]: Added some info. We should double check to see if we need `known_third_party`. Maintaining this list seems like it could become a lot of overhead.

File: tests/test_eval_detection.py [hunk 4]
```diff
@@ -215,8 +216,26 @@ def test_accumulate() -> None:
         cls_to_accum["VEHICLE"]
         == np.array(
             [
-                [1.0, 1.0, 1.0, 1.0, expected_ATE, expected_ASE, expected_AOE, expected_AP],
-                [1.0, 1.0, 1.0, 1.0, expected_ATE, expected_ASE, expected_AOE, expected_AP],
+                [
```
Comment (johnwlambert) [inline tests/test_eval_detection.py:219]: can we remove these changes from this PR, Ben?

**Comment by johnwlambert** [comment]
Ben, what are the advantages of pyproject.toml over our current repo setup?
